test/e2e/e2e_suite_test.go

```diff
@@ -103,26 +101,40 @@ var _ = SynchronizedBeforeSuite(func() []byte {
 	if e2eConfig.GetVariable("DEPLOY_CERT_MANAGER") != "false" {
 		// Install cert-manager
 		By("Installing cert-manager")
-		cmd := exec.Command("cmctl", "check", "api")
-		_, err := cmd.CombinedOutput()
+		err := checkCertManagerAPI(clusterProxy)
 		if err != nil {
-			cmd = exec.Command("cmctl", "x", "install")
-			output, err := cmd.CombinedOutput()
-			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("%s", output))
-			By("Checking that the cert-manager API is available")
-			cmd = exec.Command("cmctl", "check", "api")
-			output, err = cmd.CombinedOutput()
-			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("%s", output))
+			cmVersion := e2eConfig.GetVariable("CERT_MANAGER_VERSION")
+			err = installCertManager(ctx, clusterProxy, cmVersion)
+			Expect(err).NotTo(HaveOccurred())
+			// wait for all cert-manager deployments to be available
+			cmDeploymentNames := []string{
+				"cert-manager",
+				"cert-manager-webhook",
+				"cert-manager-cainjector",
+			}
+			for _, name := range cmDeploymentNames {
+				cmDeployment := &v1.Deployment{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      name,
+						Namespace: "cert-manager",
+					},
+				}
+				framework.WaitForDeploymentsAvailable(ctx, framework.WaitForDeploymentsAvailableInput{
```

Comment:
This is unfortunately not enough. The webhook is not always available and registered as fast as the deployment becomes available.
Could you add a function that waits and retries the same way as clusterctl and cmctl does it? It should try to create a self signed issuer and certificate again and again until it succeeds.

Reply:
Ah, okay. I misunderstood your idea before xD